### ISSUE TITLE
[Bug]: SocketIO event regression, AgentStateChangeObservation is double emitted and emitted out of sequence

### ISSUE THREAD

**Issue by diwu-sf** [All-Hands-AI/OpenHands#5661]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Describe the bug and reproduction steps

This is a regression in the `listen_socket.py` change from https://github.com/All-Hands-AI/OpenHands/pull/5632/files

the bug is this change:
![image](https://github.com/user-attachments/assets/32880b21-a328-444e-a878-ae4376828b38)

Here are the bugs:
1. In the event stream, the sequence of state change is LOADING -> INIT -> ..., this code would capture LOADING as the `agent_state_changed` event and emit it at the end, causing the frontend to think the app is still in LOADING state when it should be in INIT state
2. This will also double emit any INIT events, unnecessary
3. the literal `init` should not be used, instead `AgentState.INIT` constant should be used

I think this section should be fixed as something like:
```py
        elif isinstance(event, AgentStateChangedObservation):
            if event.agent_state == AgentState.INIT:
                await sio.emit('oh_event', event_to_dict(event), to=connection_id)
            agent_state_changed = event
        else:
            await sio.emit('oh_event', event_to_dict(event), to=connection_id)
    if agent_state_changed:
        await sio.emit('oh_event', event_to_dict(agent_state_changed), to=connection_id)
```

### OpenHands Installation

Docker command in README

### Operating System

None

**Comment by amanape** [comment]
Some background: The WS on the frontend is only set to active once receiving this event. This change was made so that it received the same event and is able to appropriately set the WS state on refreshes as well.

Other than that, the changes do make sense to me. I will try to take a look at it later today, but it seems you've already done all the steps except submit a PR 😃 

Why not give it a shot and I'll review it?

**Comment by diwu-sf** [comment]
Let's ask openhands and see if it can pick up on my implementation

**Comment by openhands-agent** [comment]
[OpenHands](https://github.com/All-Hands-AI/OpenHands) started fixing the issue! You can monitor the progress [here](https://github.com/All-Hands-AI/OpenHands/actions/runs/12388420893).

**Comment by openhands-agent** [comment]
A potential fix has been generated and a draft PR #5663 has been created. Please review the changes.
